CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+# Contributing to ISOBUS++
+
+We accept public contributions that follow our code of conduct, pass all automated pre-merge checks, and pass a manual code review by a repo maintainer.
+
+## What Are The Requirements?
+
+Our style rules and PR reviews are based loosely on Autosar and MISRA suggestions and try to ensure the highest quality possible.
+
+* Contributions must follow the style defined in our `.clang-format` file. You can ensure you pass this check by running `find . -iname *.hpp -o -iname *.cpp | xargs clang-format -i` at the root of the repo.
+* Contributions should follow these additional style requirements, which will be checked in code reviews.
+	* Function names `snake_case`
+	* Variables `camelCase`
+	* Constant values `CAPITALIZED_SNAKE`
+	* Constants on the left in `==` and `!=` checks. Like this: `if (5 == value)` NOT `if (value == 5)`. This is to prevent accidentally omitting an `=` in the operator and creating a runtime bug.
+* Doxygen should compile with no warnings when run at the root of the project with the command `doxygen doxyfile`
+* Copyright notice must be included in each file
+* Absolutely no code shall be added that is under a more strict license than MIT or which has not had conditions met to be distributed under our license
+* Builds must pass the compilation github action
+* `NULL` should not be used when `nullptr` can be used
+* Prefer C++ over C, though exceptions may be granted if needed
+* C++11 or earlier is required to help provide the maximum compatibility to all modern compilers, especially for embedded ones that do not support the most up-to-date versions of the standard, such as ARM Compiler 5 and 6.
```

Comment:
ARM Compiler V6 fully supports C++14. May be somewhat freeing to drop support for V5 which doesn't even fully support C++11.

Reply:
Yeah, I can get on-board with that. 5 is technically EOL I think so I am ok with raising it to 14